### PR TITLE
UP-4951: Redirect Guest to CAS

### PR DESCRIPTION
**Pull request by bjagg**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]
-   [x] tests are included
-   [x] documentation is changed or added

##### Description of change

Configure 'Redirect Guest to CAS' with a single property value.

https://issues.jasig.org/browse/UP-4951

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

**Comment by bjagg** [comment]
Documentation still needs an update.

**Comment by drewwills** [comment]
I'm glad to have something new in this area.  I'm planning on taking a closer look today.

File: docs/implement/authentication/redirect-guest-to-cas.md [hunk 1]
```diff
@@ -1,7 +1,17 @@
...
 
 An occasionally feature request is for uPortal to only support authenticated users,
-skipping a guest experience all together.
+skipping a guest experience all together. There are two solutions depending on which version of uPortal is in use.
+
+## uPortal 5+
+
+To enable the redirect, set the following property to true in `uPortal.properties`:
+
+```properties
```
Comment (ChristianMurphy) [inline docs/implement/authentication/redirect-guest-to-cas.md]: :star: thanks for using a language flag on the code fence.

File: docs/implement/authentication/redirect-guest-to-cas.md [hunk 2]
```diff
@@ -1,7 +1,17 @@
 # Redirecting Guests to CAS for Sign-In
 
 An occasionally feature request is for uPortal to only support authenticated users,
-skipping a guest experience all together.
+skipping a guest experience all together. There are two solutions depending on which version of uPortal is in use.
```
Comment (drewwills) [inline docs/implement/authentication/redirect-guest-to-cas.md]: @bjagg -- I'm pretty sure the repo-housed documentation within the `docs/` directory is uP5-only.  The Confluence-based manuals are for previous versions.

File: uPortal-webapp/src/main/resources/properties/contexts/securityContext.xml [hunk 3]
```diff
@@ -169,4 +169,19 @@
         </property>
     </bean>
 
+    <!--
+     | Spring-managed instance of CAS AuthenticationFilter. As above, this allows us to configure
+     | CAS authentication in the preferred approach. This is only needed for 'Guest Redirect to CAS'.
+     +-->
+    <bean name="casAuthenticationFilter" class="org.apereo.portal.spring.web.BypassFilterDecorator">
```
Comment (drewwills) [inline uPortal-webapp/src/main/resources/properties/contexts/securityContext.xml]: Did the name change to `OptionalFilterDecorator`?

I'm getting...

```
Caused by: java.lang.ClassNotFoundException: org.apereo.portal.spring.web.BypassFilterDecorator
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1291)
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1119)
	at org.springframework.util.ClassUtils.forName(ClassUtils.java:250)
	at org.springframework.beans.factory.support.AbstractBeanDefinition.resolveBeanClass(AbstractBeanDefinition.java:401)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doResolveBeanClass(AbstractBeanFactory.java:1432)
	at org.springframework.beans.factory.support.AbstractBeanFactory.resolveBeanClass(AbstractBeanFactory.java:1377)
	... 43 common frames omitted
```

Comment (bjagg) [reply uPortal-webapp/src/main/resources/properties/contexts/securityContext.xml]: Yep ... fixed.

File: docs/implement/authentication/redirect-guest-to-cas.md (hunk 2 above)
Comment (bjagg) [reply docs/implement/authentication/redirect-guest-to-cas.md]: Then what do you think about removing the doc?

Comment (drewwills) [reply docs/implement/authentication/redirect-guest-to-cas.md]: Exactly -- I'd say just cover the uPortal 5 way as if it's the only way.

**Comment by ChristianMurphy** [comment]
Hey @bjagg, it looks like a merge conflict has cropped up. Would you be able to resolve the conflict?

**Comment by drewwills** [comment]
@ChristianMurphy & @bjagg -- how about I just take this one from here?  (@bjagg is working on something important.)

**Comment by bjagg** [comment]
Tested and ready for any final review

**Comment by drewwills** [comment]
I'm taking a quick look.

File: docs/implement/authentication/README.md [hunk 4]
```diff
@@ -1,17 +1,18 @@
...
-uPortal supports various methods to authenticate users.
 
-+  Internal Authentication (TBD)
-+ [Redirect Guest to CAS Sign-In](redirect-guest-to-cas.md)
-+ [CAS 5 ClearPass: Credential Caching and Replay](Cas5ClearPass.md)
+uPortal supports a comprehensive list of strategies for authenticating users, including [Apereo
+Central Authentication Service][] (CAS), [Shibboleth][]/SAML, [LDAP][], and several others.
 
-## Redirecting Guests to CAS for Sign-In
+This strategies may also be mixed or combined in any way.
```
Comment (bjagg) [inline docs/implement/authentication/README.md]: These?

**Pull request merged**